src/quo2/components/code/snippet.cljs

```diff
@@ -1,175 +1,110 @@
+                  (cond-> {:weight :code
+                           :size   :paragraph-2
+                           :style  (style/text-style (get-in node [:properties :className]))}
+                    last-line? (assoc :number-of-lines 1))]
                  (render-nodes children))
            ;; Remove newlines as we already render each line separately.
-           (-> value string/trim-newline)))
+           (string/trim-newline value)))
        nodes))
 
+(defn- line
+  [{:keys [line-number line-number-width]} children]
+  [rn/view {:style style/line}
+   [rn/view {:style (style/line-number line-number-width)}
+    [text/text
+     {:style  (style/text-style ["line-number"])
+      :weight :code
+      :size   :paragraph-2}
+     line-number]]
+   children])
+
 (defn- code-block
   [{:keys [rows line-number-width]}]
-  [into [:<>]
+  [rn/view
    (->> rows
         (render-nodes)
-        ;; Line numbers
-        (map-indexed (fn [idx row]
-                       (conj [rn/view {:style {:flex-direction :row}}
-                              [rn/view
-                               {:style {:width        line-number-width
-                                        ;; 8+12 margin
-                                        :margin-right 20}}
-                               [text/text
-                                {:weight :code
-                                 :size   :paragraph-2
-                                 :style  (text-style ["line-number"])}
-                                (inc idx)]]]
-                             row))))])
+        (map-indexed (fn [idx row-content]
+                       [line
+                        {:line-number       (inc idx)
+                         :line-number-width line-number-width}
+                        row-content]))
+        (into [:<>]))])
 
-(defn- native-renderer
-  []
-  (let [text-height (reagent/atom nil)]
-    (fn [{:keys [rows max-lines on-copy-press]}]
-      (let [background-color      (colors/theme-colors
-                                   colors/white
-                                   colors/neutral-80-opa-40)
-            background-color-left (colors/theme-colors
-                                   colors/neutral-5
-                                   colors/neutral-80)
-            border-color          (colors/theme-colors
-                                   colors/neutral-20
-                                   colors/neutral-80)
-            rows                  (bean/->clj rows)
-            font-scale            (:font-scale (rn/use-window-dimensions))
-            max-rows              (or max-lines (count rows)) ;; Cut down on rows to process.
-            max-line-digits       (-> rows count (min max-rows) str count)
-            ;; ~ 9 is char width, 18 is width used in Figma.
-            line-number-width     (* font-scale (max 18 (* 9 max-line-digits)))
-            max-text-height       (some-> max-lines
```

Comment:
We're no longer using `max-text-height` or any other height related calculation, this was creating a bug where if you said you want to show `n` lines from a fragment of code, sometimes it was showing less than `n` depending on the screen size. Now it's fixed

The problem was due to this height calculation, this height is supposing the  height of a drawn line is constant, but on smaller screens, a line could be drawn using a bigger size. e.g.

The folowing code:
```clojure
#_line-1 (let [a nil]
#_line-2  [button/button {:override-background-color (style/button-background-color)} :main-icons/copy]
#_line-3  nil)
```
Has 3 lines, this code will calculate a specific height, but if the phone screen is not wider enough to show the second line in one single drawn line, the view won't be shown as expected. Depending on screen's width, it cold be shown as:
```clojure
#_line-1 (let [a nil]
#_line-2  [button/button {:override-background-color (style/button-
#_line-2 background-color)} :main-icons/copy]
#_line-3  nil)
```
or
```clojure
#_line-1 (let [a nil]
#_line-2  [button/button {:override-background
#_line-2 -color (style/button- background-color
#_line-2 )} :main-icons/copy]
#_line-3  nil)
```

causing the user sees less lines than requested